classes/task/merge_user_accounts.php

```diff
@@ -0,0 +1,194 @@
+        $removeuserfield = $record->removeuserfield;
+        $removeuservalue = $record->removeuservalue;
+        // Verify user to remove.
+        $usertoremove = $DB->get_records(merge_request::TABLE_USERS, [$removeuserfield => $removeuservalue]);
+        if (is_null($record->removeuserid)) {
+            $this->removeuserid = $this->find_user_id_or_fail($record->removeuserfield, $record->removeuservalue);
+        }
+        // Verify user to keep.
+        $usertokeep = $DB->get_records(merge_request::TABLE_USERS, [$keepuserfield => $keepuservalue]);
+        if (is_null($record->keepuserid)) {
+            $this->keepuserid = $this->find_user_id_or_fail($record->keepuserfield, $record->keepuservalue);
+        }
+    }
+    /**
+     * Function for updating number of retries.
+     */
+    private function update_retries_in_table(int $idrecord,
+                                             int $retries): void {
+        global $DB;
+        $DB->update_record(
+            merge_request::TABLE_MERGE_REQUEST,
+            (object)[
+                'id' => $idrecord,
+                'retries' => $retries,
+                'timecompleted' >= time(),
+            ]
+        );
+    }
+    /**
+     * Function for updating removeuserid.
+     */
+    private function update_removeuserid_in_table(int $idrecord,
+                                                        int $removeuserid): void {
+        global $DB;
+        $DB->update_record(
+            merge_request::TABLE_MERGE_REQUEST,
+            (object)[
+                'id' => $idrecord,
+                'removeuserid' => $removeuserid
+            ]
+        );
+    }
+    /**
+     * Function for updating keepuserid.
+     */
+    private function update_keepuserid_in_table(int $idrecord,
+                                                int $keepuserid) {
+        global $DB;
+        $DB->update_record(
+            merge_request::TABLE_MERGE_REQUEST,
+            (object)[
+                'id' => $idrecord,
+                'keepuserid' => $keepuserid
+            ]
+        );
+    }
+    /**
+     * Function to find user id or fail.
+     */
+    protected function find_user_id_or_fail(int $mergerequestid, string $userfield, string $uservalue): int {
+        global $DB;
+        $users = $DB->get_records(merge_request::TABLE_USERS, [$userfield => $uservalue]);
+        if (count($users) == 0) {
+            throw new moodle_exception(get_string('cannotfinduser', 'tool_mergeusers',
+                                        (object)['userfield' => $userfield, 'uservalue' => $uservalue]));
+        }
+        if (count($users) > 1) {
+            throw new moodle_exception(get_string('toomanyusers', 'tool_mergeusers',
+                                        (object)['userfield' => $userfield, 'uservalue' => $uservalue]));
+        }
+        $user = reset($users);
+        $this->update_removeuserid_in_table($mergerequestid , $user->id);
```

Comment:
We are in an error here: this method is used from both user to keep and user to remove, but here we ONLY update the removeuserid. 

This method is responsible to find the user id, so this method should end with `return $user->id;`. From the client side will use the returned id to invoke the corresponding method, to update the remove user id or the user to keep.